code/game/objects/items/explosives/mine.dm

```diff
@@ -1,277 +1,1200 @@
 	. = ..()
 	alpha = armed ? 50 : 255
 
-/obj/item/explosive/mine/anti_tank/trigger_explosion()
+/obj/item/mine/incendiary
+	name = "napalm mine"
+	desc = "Incendiary mine variant with a napalm-based formula. Very sticky."
+	icon_state = "incendiary"
+	detonation_message = "hisses, releasing an inferno."
+	detonation_sound = 'sound/weapons/guns/fire/tank_flamethrower.ogg'
+	range = 3
+	disarm_delay = 2 SECONDS
+	undeploy_delay = 1 SECONDS
+	deploy_delay = 1 SECONDS
+	fire_range = 4
+	fire_intensity = 10 SECONDS
+	fire_duration = 4 SECONDS
+	fire_damage = 10
+	fire_stacks = 10
+	fire_color = "green"
+	mine_features = MINE_DISCERN_LIVING|MINE_ELECTRONIC|MINE_VOLATILE_DAMAGE|MINE_VOLATILE_FIRE|MINE_VOLATILE_EXPLOSION
+
+/* Improvised explosives - You craft them */
+/obj/item/mine/ied
+	name = "pipe mine"
+	desc = "Cobbled together from a metal pipe, gunpowder, rods, cables, and a proximity sensor. The people's mine."
+	icon_state = "pipe"
+	detonation_message = "clicks and rattles."
+	detonation_sound = 'sound/machines/triple_beep.ogg'
+	range = 2
+	angle = 360
+	detonation_delay = 0.5 SECONDS
+	disarm_delay = 1 SECONDS
+	undeploy_delay = 2 SECONDS
+	deploy_delay = 2 SECONDS
+	shrapnel_type = /datum/ammo/bullet/claymore_shrapnel/nail
+	shrapnel_range = 4
+	//I know they technically are electronic with the proximity sensor but it's probably more fun if not EMP'able since they're practically junk
+	mine_features = MINE_INTERRUPTIBLE|MINE_VOLATILE_DAMAGE|MINE_VOLATILE_FIRE|MINE_VOLATILE_EXPLOSION
+
+	///Stages of assembly; see top of file for the flags
+	var/assembly_steps_completed = NONE
+	///How much gunpowder is needed for the mine to be operational
+	var/gunpowder_amount_required = 20
+	///How much gunpowder is in the mine
+	var/gunpowder_amount = 0
+	///How many rods are needed for the mine to be operational
+	var/rods_amount_required = 4
+	///How many rods have been added to this mine
+	var/rods_amount = 0
+
+/obj/item/mine/ied/examine(mob/user)
+	. = ..()
+	if(gunpowder_amount < gunpowder_amount_required)
+		. += "Currently has [span_bold("[gunpowder_amount]")] out of [span_bold("[gunpowder_amount_required]")] gunpowder."
+	if(rods_amount < rods_amount_required)
+		. += "Currently has [span_bold("[rods_amount]")] out of [span_bold("[rods_amount_required]")] rods."
+
+/obj/item/mine/ied/update_icon_state()
+	. = ..()
+	cut_overlays()
+	//Only add the wires if not deployed, don't need the overlay for that
+	if(CHECK_BITFIELD(assembly_steps_completed, IED_WIRED) && !anchored)
+		var/image/wires_overlay = image(icon, src, "wires")
+		add_overlay(wires_overlay)
+
+/obj/item/mine/ied/setup(mob/living/user)
+	if(!CHECK_MULTIPLE_BITFIELDS(assembly_steps_completed, IED_FINISHED) || gunpowder_amount < gunpowder_amount_required || rods_amount < rods_amount_required)
+		balloon_alert(user, "Not finished!")
+		return FALSE
+	return ..()
+
+/obj/item/mine/ied/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(isscrewdriver(I))
```

Comment:
...yeah duh. This gets handled by `melee_attack_chain` into `tool_attack_chain`, 99% sure. You don't need to do this.